From 9c4d33c877a974e85b9dd5a93baad6d60499a601 Mon Sep 17 00:00:00 2001
From: Georg Koppen <gk@torproject.org>
Date: Wed, 30 Jan 2019 15:38:44 +0000
Subject: [PATCH] Bug 28716: Optionally omit timestamp in PE headers


diff --git a/MinGW/Driver.cpp b/MinGW/Driver.cpp
index d682baf59..afdd26541 100644
--- a/MinGW/Driver.cpp
+++ b/MinGW/Driver.cpp
@@ -170,6 +170,9 @@ bool mingw::link(ArrayRef<const char *> ArgsArr, raw_ostream &Diag) {
       Args.getLastArgValue(OPT_m) != "arm64pe" && !Args.hasArg(OPT_dynamicbase))
     Add("-dynamicbase:no");
 
+  if (Args.hasArg(OPT_no_insert_timestamp))
+    Add("-timestamp:0");
+
   if (Args.hasFlag(OPT_gc_sections, OPT_no_gc_sections, false))
     Add("-opt:ref");
   else
diff --git a/MinGW/Options.td b/MinGW/Options.td
index 948faa687..8e44b321a 100644
--- a/MinGW/Options.td
+++ b/MinGW/Options.td
@@ -20,6 +20,8 @@ def l: JoinedOrSeparate<["-"], "l">, MetaVarName<"<libName>">,
 def m: JoinedOrSeparate<["-"], "m">, HelpText<"Set target emulation">;
 def map: S<"Map">, HelpText<"Output a linker map">;
 def map_eq: J<"Map=">, Alias<map>;
+def no_insert_timestamp: F<"no-insert-timestamp">,
+    HelpText<"Don't include PE header timestamp">;
 def no_whole_archive: F<"no-whole-archive">,
     HelpText<"No longer include all object files for following archives">;
 def large_address_aware: Flag<["--"], "large-address-aware">,
-- 
2.20.1

